pkg/beholder/message.go

```diff
@@ -222,8 +232,35 @@ func NewMetadata(attrs Attributes) *Metadata {
 	return m
 }
 
-func (m *Metadata) Validate() error {
+// validDomainAndEntityRegex allows for alphanumeric characters and ._-
+var validDomainAndEntityRegex = regexp.MustCompile(`^[a-zA-Z0-9._-]+$`)
```

Comment:
I wonder if this is worth optimizing since it will be such high volume?

Reply:
I think this regex would just be compiled once and then reused as it's at the package level, which should help with high volume

or is there a different optimization you had in mind?

Reply:
This won't run in the playground, but if you run it locally it performs a rough benchmark: https://go.dev/play/p/oZngGcqQXqC

Regex is relatively slow, but it may not be significant:
```
Regex: 3.690749ms
ASCII: 64.001µs
Unicode: 327.251µs
```